### DISCUSSION TITLE
Operation of Text-Ts and Binary-Ts

### DISCUSSION THREAD

**Discussion by rma-rripken** [USACE/cwms-data-api#540] [General]
Text-Ts and Binary-Ts can either exist on their own or be annotations on another timeseries.

I believe:
      When text-ts exist on their own the parameter will be "text".
      When binary-ts exist on their own the parameter will be "binary".
      In all the other cases the text-ts or binary-ts are annotations of something else.

When a text-ts or binary-ts is an annotation of another timeseries must that other timeseries exist before the text-ts or binary-ts is created?
Must the interval offset be defined?  ( Seems like it )
Must it have data in it?  ( Regular text-ts doesn't seem to care but standard text-ts seems like it does )

Are there any requirements about the time window extents of the source (assume numeric) timeseries?  Do there need to be numeric values at the time-periods of the text-ts ?  Do data points have to exist before?  After?

What should happen to the text-ts annotations if the numeric timeseries is deleted?  Or if the series isn't deleted but all the data points are? 

Is there documentation of the desired behavior you can point me at?
I can debug into some of the issues but first I need to understand how its supposed to work.

**Comment by adamkorynta** [comment]
I'd like to add another clarification: are the date versions for the annotated text/binary TS linked with the date versions for the numerical TS?

**Reply by MikeNeilson** [reply]
Good question, one would think so but I'd have to look. Might be a better question for Perryman on Tuesday (it's travelling today.)
